### ISSUE TITLE
Deprecate HPC/SLURM specific checkpointing

### ISSUE THREAD

**Issue by tarepan** [PyTorchLightning/pytorch-lightning#5373]
## 🚀 Feature
Deprecate HPC-specific checkpointing in [`CheckpointConnector`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/checkpoint_connector.py).  

### Motivation
`CheckpointConnector` has HPC/SLURM-specific save-load system for auto-resubmit ([doc](https://pytorch-lightning.readthedocs.io/en/stable/slurm.html#wall-time-auto-resubmit)).  
Now that auto-resubmit is supported in normal restore process (#4402), HPC auto-resubmit also can be handled by this general process.  
In my opinion, the HPC-specific auto-submit ended its historical role.  
By deprecating this specific system, `CheckpointConnector` can be refactored so simple and become easy to maintain.  

### Pitch
Deprecate `hpc_save` & `hpc_load`, which use **`hpc_ckpt_{ckpt_number}.ckpt` name convention** for auto-resume/resubmit.  
Use general auto-resubmit system, which attempt to use `last.ckpt` automatically, for SLURM.  

#### Backward compatibility
The deprecation break previously-generated checkpoint for auto-resubmit.  
But auto-resubmit checkpoint is, in general, used within short-term.  
In other words, the checkpoint is ephemeral.  

And `hpc_save` and `hpc_load` are internal method (no public API in docs).  

In this point of views, in my opinion, we can deprecate (internal) old system without deprecation warning/term.

**Comment by tarepan** [comment]
Are there any opinion, contributors?  
If core contributors agree, I will make PR for this issue.

**Comment by carmocca** [comment]
The idea looks good to me overall.

This issue is pretty much a duplicate of https://github.com/PyTorchLightning/pytorch-lightning/issues/6204

Closing this one even though it was created first since it contains less activity
